hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -279,24 +289,22 @@ private void populateResultFromNearCache(Collection<?> keys, List<Object> result
         }
     }
 
-    private Map<Object, Long> getNearCacheReservations(Collection<?> nearCacheKeys, Map<Object, Data> keyMap) {
-        Map<Object, Long> reservations = createHashMap(nearCacheKeys.size());
-        for (Object key : nearCacheKeys) {
-            Data keyData = serializeKeys ? (Data) key : keyMap.get(key);
-            long reservationId = tryReserveForUpdate(key, keyData);
+    private void createNearCacheReservations(Collection<?> dataKeys, Map<Object, Long> reservations) {
```

Comment:
That's true, you can pass in anything. But this line would fail when the element type is not assignable to `Data`: 
`for (Object key : dataKeys) {
            long reservationId = tryReserveForUpdate(key, (Data) key);`